papaparse.js

```diff
@@ -1457,6 +1457,34 @@ License: MIT
 			if (!input)
 				return returnable();
 
+			// Rename headers if there are duplicates
+			if (config.header)
+			{
+				let firstLine = input.split(newline)[0]
+				let separator = '_';
+				let headers = firstLine.split(delim) 
+				let header_map = []; 
+				let header_count = {};
+
+				for (var i in headers){
+					if (!header_map.includes(headers[i])){
+						header_map.push(headers[i])
+					}
+					else{
+						if (!(headers[i] in header_count)){
+							header_count[headers[i]] = 1
+						}
+						else{
+							header_count[headers[i]] +=1
+						}
+						header_map.push(headers[i]+separator+header_count[headers[i]])
```

Comment:
I will prefer to have a single line that pushes to `header_map`.
So you just need to have a variable name `header_name` that is initialized with the same value as `header`

Then if `header` is already in `header_map+  you just save the count and update the variable name.